bigchaindb/core.py

```diff
@@ -485,3 +487,30 @@ def get_unvoted_blocks(self):
             unvoted.pop(0)
 
         return unvoted
+
+    def block_election_status(self, block):
+        """
+        Tallies the votes on a block, and returns the status: valid, invalid, or undecided
+        """
+        n_voters = len(block['block']['voters'])
+        vote_cast = [vote['vote']['is_block_valid'] for vote in block['votes']]
+        vote_validity = [self.consensus.verify_vote_signature(block, vote) for vote in block['votes']]
+
+        # element-wise product of stated vote and validity of vote
+        vote_list = list(map(operator.mul, vote_cast, vote_validity))
+
+        # validate votes here
+        n_valid_votes = sum(vote_list)
+        n_invalid_votes = len(vote_list) - n_valid_votes
+
+        # The use of ceiling and floor is to account for the case of an
+        # even number of voters where half the voters have voted 'invalid'
+        # and half 'valid'. In this case, the block should be marked invalid
+        # to avoid a tie. In the case of an odd number of voters this is not
+        # relevant, since one side must be a majority.
+        if n_invalid_votes >= math.ceil(n_voters / 2):
+            return 'invalid'
```

Comment:
would be nice to put `invalid`, `valid`, and `undecided` under `Bigchain.BLOCK_INVALID = 'invalid'`, `Bigchaindb.BLOCK_VALID = 'valid'', so it's easier for a developer to use the return value of the method. Also, IDEs and editors with code completion will make use of it.